test/security/testrunners/filepermissiontestrunner.go

```diff
@@ -0,0 +1,108 @@
+package testrunners
+
+import (
+	"github.com/aws/amazon-cloudwatch-agent-test/internal/common"
+	"github.com/aws/amazon-cloudwatch-agent-test/internal/rule"
+	"github.com/aws/amazon-cloudwatch-agent-test/test/status"
+	"github.com/aws/amazon-cloudwatch-agent-test/test/test_runner"
+	"log"
+	"strings"
+	"time"
+)
+
+type FilePermissionTestRunner struct {
+	test_runner.BaseTestRunner
+}
+
+var _ test_runner.ITestRunner = (*FilePermissionTestRunner)(nil)
+
+const agentConfigPath = "/opt/aws/amazon-cloudwatch-agent/bin/config.json"
+const agentConfigOnlyRootRead = "-rw-r--r-- 1 root root "
+
+var (
+	onlyRootReadExactMatchRule = rule.Rule{
+		Conditions: []rule.ICondition{
+			&rule.ExactMatch{ExpectedValue: agentConfigOnlyRootRead},
+		},
+	}
+)
+
+var testCases = map[string]rule.Rule{
+	agentConfigPath: onlyRootReadExactMatchRule,
+}
+
+var actualPermissions = make(map[string]string)
+
+func (m *FilePermissionTestRunner) Validate() status.TestGroupResult {
+	testResults := make([]status.TestResult, len(testCases))
+
+	count := 0
+	for k, v := range testCases {
+		testResults[count] = m.validatePermissions(k, v)
+		count++
+	}
+
+	return status.TestGroupResult{
+		Name:        m.GetTestName(),
+		TestResults: testResults,
+	}
+}
+
+func (m *FilePermissionTestRunner) GetTestName() string {
+	return "FilePermission"
+}
+
+func (m *FilePermissionTestRunner) GetAgentConfigFileName() string {
+	return "minimum_config.json"
+}
+
+func (m *FilePermissionTestRunner) GetMeasuredMetrics() []string {
+	return []string{}
+}
+
+func (m *FilePermissionTestRunner) GetAgentRunDuration() time.Duration {
+	return 1 * time.Minute
+}
+
+func (m *FilePermissionTestRunner) SetupAfterAgentRun() error {
+	for k, _ := range testCases {
+		p, err := getFilePermission(k)
+		if err != nil {
+			return err
+		}
+
+		log.Printf("file perission is %s", p)
+		actualPermissions[k] = p
+	}
+
+	return nil
+}
+
+func (m *FilePermissionTestRunner) validatePermissions(fileTestedPath string, rule rule.Rule) status.TestResult {
+	log.Printf("Validating Permission for  %v", fileTestedPath)
+
+	testResult := status.TestResult{
+		Name:   fileTestedPath,
+		Status: status.FAILED,
+	}
+
+	if !rule.Evaluate(actualPermissions[fileTestedPath]) {
+		return testResult
+	}
+
+	testResult.Status = status.SUCCESSFUL
+	return testResult
+}
+
+func getFilePermission(filePath string) (string, error) {
```

Comment:
Maybe [leverage from these functions instead of looking for `ls` ](https://github.com/aws/amazon-cloudwatch-agent-test/blob/main/filesystem/unix_permission.go#L17-L50) (e.g `CheckFileRights`)
```
// CheckFileRights check that the given file path has been protected by the owner.
// If the owner is changed, they need at least the sudo permission to override the owner.
```

Reply:
:o i thought I looked if this exists already

Reply:
You could also consider just doing `os.Stat` and getting the `FileMode` from the `FileInfo` https://pkg.go.dev/io/fs#FileInfo

You could then validate it against 0664 or whatever the expected permissions are.

Reply:
I'm fine with either. We'll need something to check windows too but I'm indifferent about this for now

Reply:
Apparently, it's portable and can be used on Windows too. https://medium.com/@MichalPristas/go-and-file-perms-on-windows-3c944d55dd44

Reply:
oh yay